clients/fides-js/src/lib/initialize.ts

```diff
@@ -317,14 +319,21 @@ export const initialize = async ({
       shouldInitOverlay = false;
     } else if (!isPrivacyExperience(effectiveExperience)) {
       fetchedClientSideExperience = true;
-      // If no effective PrivacyExperience was pre-fetched, fetch one now from
-      // the Fides API using the current region string
-      effectiveExperience = await fetchExperience(
-        fidesRegionString,
-        options.fidesApiUrl,
-        options.debug,
-        cookie.identity.fides_user_device_id
-      );
+      // If no effective PrivacyExperience was pre-fetched, fetch one using the current region string
+      if (options.apiOptions?.getPrivacyExperienceFn) {
+        effectiveExperience = await customGetPrivacyExperience(
```

Comment:
I don't feel super strongly about this, but I wonder if it would make things easier or harder if we did the check for if there's a custom function inside the relevant function call, instead of outside of it? for example:

```js
const fetchExperience = (params) => {
  if (options.apiOptions?.getPrivacyExperienceFn) {
    return await customGetPrivacyExperience(...)
  }
  // whatever we already had
}
```

that way the outside logic could stay the same, and it's just the function internals of `fetchExperience` (and `patchNoticesServed`) that is changing?

maybe that complicates other arguments? but I think it'd be nicer in the sense of if we `fetchExperience` or `patchNoticesServed` in more than one place, we are covered

Reply:
having it the way you have it now also makes things a little odd, in that there are two checks for `options.apiOptions?.getPrivacyExperienceFn`—one here, and one inside `customGetPrivacyExperience`